Fix AppVeyor failing

Currently all four VS2012 (ver 11) configurations failed during compilation:
~~~
C:\projects\rapidjson\test\unittest\stringbuffertest.cpp(86): error C2338:  [C:\projects\rapidjson\Build\VS\test\unittest\unittest.vcxproj]
C:\projects\rapidjson\test\unittest\stringbuffertest.cpp(87): error C2338:  [C:\projects\rapidjson\Build\VS\test\unittest\unittest.vcxproj]
C:\projects\rapidjson\test\unittest\stringbuffertest.cpp(95): error C2338:  [C:\projects\rapidjson\Build\VS\test\unittest\unittest.vcxproj]
~~~

--- a/test/unittest/stringbuffertest.cpp
+++ b/test/unittest/stringbuffertest.cpp
@@ -83,8 +83,11 @@ TEST(StringBuffer, Traits) {
 
     static_assert(!std::is_nothrow_constructible<StringBuffer>::value, "");
     static_assert(!std::is_nothrow_default_constructible<StringBuffer>::value, "");
+
+#if !defined(_MSC_VER) || _MSC_VER >= 1800
     static_assert(!std::is_nothrow_copy_constructible<StringBuffer>::value, "");
     static_assert(!std::is_nothrow_move_constructible<StringBuffer>::value, "");
+#endif
 
     static_assert( std::is_assignable<StringBuffer,StringBuffer>::value, "");
 #ifndef _MSC_VER
@@ -92,7 +95,10 @@ TEST(StringBuffer, Traits) {
 #endif
     static_assert( std::is_move_assignable<StringBuffer>::value, "");
 
-    static_assert(!std::is_nothrow_assignable<StringBuffer,StringBuffer>::value, "");
+#if !defined(_MSC_VER) || _MSC_VER >= 1800
+    static_assert(!std::is_nothrow_assignable<StringBuffer, StringBuffer>::value, "");
+#endif
+
     static_assert(!std::is_nothrow_copy_assignable<StringBuffer>::value, "");
     static_assert(!std::is_nothrow_move_assignable<StringBuffer>::value, "");
 

